tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/contact/ContactAddedRabbitMqMessage.java

```diff
@@ -0,0 +1,21 @@
+package com.linagora.tmail.contact;
+
+import java.nio.charset.StandardCharsets;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public record ContactAddedRabbitMqMessage(String bookId, String bookName, String contactId,
+                                          String userId, JCardObject vcard) {
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    static ContactAddedRabbitMqMessage fromJSON(byte[] jsonBytes) {
+        String message = new String(jsonBytes, StandardCharsets.UTF_8);
+        try {
+            return objectMapper.readValue(message, ContactAddedRabbitMqMessage.class);
```

Comment:
```suggestion
        try {
            return objectMapper.readValue(jsonBytes, ContactAddedRabbitMqMessage.class);
```

?

Reply:
How about the encoding?

Reply:
Looked into the source code, they use an automatic encoding detection algorithm. It seems to be performant as it only checks some number of prefix bits to make the prediction.